### PR TITLE
cat: Put splice code in separate file, handle more failures

### PR DESCRIPTION
**Pull request by ArniDagur**
Currently, if the second call to splice fails, or if pipe creation fails, cat will not attempt to fall back to normal read/write. This was an oversight on my part. This PR fixes that.

### CODE REVIEW

**Comment by sylvestre** [comment]
I guess it is too hard to write a test for this?

**Comment by ArniDagur** [comment]
It's pretty much impossible to test the `pipe()` failure unless we use something like seccomp, but I've added tests for the second splice failing (when stdout is a file opened in append mode).

**Comment by sylvestre** [comment]
could you please fix the conflict? sorry

File: tests/by-util/test_cat.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 use crate::common::util::*;
+use std::fs::OpenOptions;
```
Comment (sylvestre) [inline tests/by-util/test_cat.rs:2]: could you please fix the warnings here?

**Comment by ArniDagur** [comment]
@sylvestre Could you take another look? Thanks.

**Comment by sylvestre** [comment]
thanks!

**Pull request merged**